iml-gui/crate/src/page/partial/header.rs

```diff
@@ -1,45 +1,83 @@
 use crate::{
-    generated::css_classes::C,
-    Model, Msg, Page,
-    Visibility::{self, *},
-    WatchState,
+    components::font_awesome, generated::css_classes::C, Model, Msg, Page,
+    Visibility::*,
 };
-use seed::{dom_types::Attrs, prelude::*, *};
+use seed::{prelude::*, *};
 
-fn font_awesome<T>(icon_name: &str) -> Node<T> {
-    svg![
-        class![C.fill_current, C.h_6, C.w_6, C.mr_3, C.inline],
-        r#use![attrs! {
-            At::Href => format!("sprites/solid.svg#{}", icon_name),
-        }]
-    ]
+fn menu_icon<T>(icon_name: &str) -> Node<T> {
+    font_awesome(class![C.h_6, C.w_6, C.mr_3, C.inline], icon_name)
 }
 
 fn nav_configure_dropdown(open: bool) -> Node<Msg> {
     if !open {
         return seed::empty();
     }
 
+    let cls = class![
+        C.my_2,
+        C.px_4,
+        C.py_2,
+        C.bg_menu_active,
+        C.block,
+        C.cursor_pointer,
+        C.hover__text_white,
+    ];
+
     div![
         class![
             C.lg__absolute,
             C.lg__border,
+            C.lg__w_full,
+            C.lg__left_0,
+            C.border_gray_600,
             C.mt_px,
-            C.text_gray_300,
+            C.text_gray_500,
             C.rounded,
-            C.p_6,
-            C.bg_menu
+            C.bg_menu,
+            C.p_4,
+            C.z_20
         ],
         style! { "top" => "110%" },
         ul![
-            li![a!["Servers"]],
-            li![a!["Power Control"]],
-            li![a!["Filesystems"]],
-            li![a!["HSM"]],
-            li![a!["Storage"]],
-            li![a!["Users"]],
-            li![a!["Volumes"]],
-            li![a!["MGTs"]]
+            li![a![
+                &cls,
+                "Servers",
+                attrs! {
+                    At::Href => Page::Server.to_href(),
+                },
+            ]],
+            li![
+                a![&cls, "Power Control"],
+                attrs! {
+                    At::Href => Page::PowerControl.to_href(),
+                },
+            ],
+            li![
+                a![&cls, "Filesystems"],
+                attrs! {
+                    At::Href => Page::Filesystem.to_href(),
+                },
+            ],
+            li![a![&cls, "HSM"]],
+            li![a![&cls, "Storage"]],
+            li![
+                a![&cls, "Users"],
```

Comment:
I noticed that menu items and page titles differ from each other, maybe `p.to_string()` could be used for menu items too?

Reply:
Yeah, that makes sense.